source/PlayerInfo.cpp

```diff
@@ -4111,13 +4111,21 @@ void PlayerInfo::SortAvailable()
 // Visit, Complete, Fail), and remove now-complete or now-failed missions.
 void PlayerInfo::StepMissions(UI *ui)
 {
-	// Check for NPCs that have been destroyed without their destruction
-	// being registered, e.g. by self-destruct:
+	// Check for NPCs that have been destroyed without their destruction being
+	// registered, e.g. by self-destruct, or landed due to the player landing.
 	for(Mission &mission : missions)
 		for(const NPC &npc : mission.NPCs())
 			for(const shared_ptr<Ship> &ship : npc.Ships())
+			{
 				if(ship->IsDestroyed())
 					mission.Do(ShipEvent(nullptr, ship, ShipEvent::DESTROY), *this, ui);
+				else if(ship->GetSystem() == system && !ship->IsDisabled()
+					&& ship->GetStopovers().count(planet) && !ship->GetStopovers().at(planet))
+				{
+					ship->LandForever();
+					mission.Do(ShipEvent(nullptr, ship, ShipEvent::LAND), *this, ui);
+				}
```

Comment:
This is the situation when we land before an NPC that has a stopover on the planet. The NPC lands forever as if it was its final destination and the mission gets the LAND event.

If we're limiting `npc land` missions to complete only at the final destination, this should be either removed (if we want to make sure the NPC lands first for the mission to complete) or changed to use the NPC's destination (if we don't care who lands first; I believe this option would also make `npc outrun` missions fail if we land on the destination when the NPC is in the same system, even if it didn't physically land, making it pretty similar to `npc evade`).

Reply:
I'm not sure which way to go on this, honestly. Maybe if it checked against the success criteria and skipped this step for missions that didn't have `ShipEvent::LAND` as a success criterion but kept it in (with the NPC's destination) for those that did?

Reply:
Sounds fine to me